index.js

```diff
@@ -52,7 +52,7 @@ function spawnClangFormat(args, done, stdio) {
   try {
     nativeBinary = getNativeBinary();
   } catch (e) {
-    setImmediate(done.bind(e));
```

Comment:
Can you explain why you're removing the `setImmediate` call in here (in the commit desc ideally)?

Reply:
This call is replaced with `done(e)` below. I'm unsure why it was ever written with `setImmediate` or with `bind`, as the `done` callback receives the error object directly.

I've tested that this works as expected on Termux on ChromeOS, whereas the `setImmediate` version does not. Unfortunately, I don't know quite what justification to give to that change, as I don't understand the reasoning behind the original version :/

Reply:
Can you research why it was added, and what it does? With this we're of course risking that we're breaking the previous reason why this was added.

Reply:
It appears that this was originally added in 4cce2c4eed4c100bf55c67a0ac515d0566514678, which introduces a bug which is supposedly fixed in 19ccfabfc7537359182ab9b414556a327e56b81f. However, none of these versions work correctly:

```shell
$ git reset --hard 4cce2 && rm -rf bin/linux_x64 && ./index.js index.js
message is not defined$ # no newline
$ git reset --hard 19cc && rm -rf bin/linux_x64 && ./index.js index.js
$ # no output
```

Conversely, with this fix (I believe this is desired):

```shell
$ git reset --hard d8759 && rm -rf bin/linux_x64/ && ./index.js index.js
HEAD is now at d875942 Fail loudly when clang-format is unavailable
Error: This module doesn't bundle the clang-format executable for your platform. (linux_x64)
Consider installing it with your native package manager instead.

    at getNativeBinary (/home/lutzky/clang-format/index.js:182:11)
    at spawnClangFormat (/home/lutzky/clang-format/index.js:53:20)
    at main (/home/lutzky/clang-format/index.js:151:5)
    at Object.<anonymous> (/home/lutzky/clang-format/index.js:204:30)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:741:12)
```

Reply:
Thanks for digging!

For reference, the PR that introduced the `setImmediate` call was https://github.com/angular/clang-format/pull/51. The reasoning of the author for the `setImmediate` call was that without it, calling code would somewhat randomly either receive an immediate exception thrown, or an async error reported via `done`. That was indeed messy, and I think it makes sense to be consistently async here.

I think the problem is likely not with `setImmediate`, but with the `bind` call. It should have been `done.bind(this, e)`, or, arguably much more readable: `setImmediate(() => done(e));`. Can you try that as a fix?

Also, the change you make below does make sense. As is, the code is calling process.exit using an error object, which is of course nonsense (and NodeJS silently accepts it :-/).

Reply:
OK, your suggested version works as well, I've force-pushed a commit with that. PTAL